### PR TITLE
Issue #1 user can search for card

### PR DESCRIPTION
**Pull request by SharjeelSIqbal**
Feature now switches views well on both mobile and desktop
Feature implements a search feature that will search the api for keywords in card names
No issues with mobile responsiveness
Search will append card images based on keywords
Kapture 2021-07-28 at 11 28 52
Kapture 2021-07-28 at 11 28 52

### CODE REVIEW

**Comment by SharjeelSIqbal** [comment]
image
image

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+function newDeck(event) {
+  event.preventDefault();
+  $newDeck.className = 'new-deck hidden';
+  $cardCount.className = 'minor-padding column-full row space-between';
+  $searchBarDiv.className = 'colun-full row justify-center hidden';
+  $infoText.className = 'row justify-center align-center';
+  $myDeckView.className = 'row justify-center align-center';
+}
```
Comment (codingcodymiller) [inline js/main.js]: This looks like something that might be suited to making a single `switchView` function like you did in `code-journal`. If more views are necessary in the future, you don't want to need to make a new function each time.

Comment (codingcodymiller) [reply js/main.js]: I recommend reviewing `javascript-view-swapping` for a reminder on how to effectively make a single function to set the current view.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by SharjeelSIqbal** [comment]
Kapture 2021-07-28 at 15 05 39

File: js/main.js [hunk 2]
```diff
@@ -1,48 +1,33 @@
...
-$myDeck.addEventListener('click', myDeckAppears);
-$search.addEventListener('click', searchBarAppears);
-$noDeck.addEventListener('click', newDeck);
-$addCard.addEventListener('click', searchBarAppears);
-$logo.addEventListener('click', myDeckAppears);
+$myDeck.addEventListener('click', switchViewing);
+$search.addEventListener('click', switchViewing);
+$noDeck.addEventListener('click', switchViewing);
+$addCard.addEventListener('click', switchViewing);
+$logo.addEventListener('click', switchViewing);
```
Comment (codingcodymiller) [inline js/main.js]: You should add a class to all of these elements so you can `querySelectorAll` for all of them, then loop over the list and add this event listener to every item with that class. That way if you want to add another element to your list of elements capable of view swapping, you just need to add that class. I recommend a class like `view-container`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js [hunk 3]
```diff
@@ -19,15 +13,15 @@ function switchView(dataView) {
...
   event.preventDefault();
-  var dataView = event.target.getAttribute('data-view');
-  switchView(dataView);
+  var $viewSwapper = document.querySelectorAll('.view-swap');
+  for (var i = 0; i < $viewSwapper.length; i++) {
+    if ($viewSwapper[i] === event.target) {
+      var dataView = $viewSwapper[i].getAttribute('data-view');
+      switchView(dataView);
+    }
+  }
```
Comment (codingcodymiller) [inline js/main.js]: Done this way, you don't need a loop, you just need to check if the `event.target` element has the class `view-swap`. If it does, you want to call the `switchView` function with the value of that `data-view` attribute

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js [hunk 4]
```diff
@@ -13,12 +13,11 @@ function switchView(dataView) {
...
-  var $viewSwapper = document.querySelectorAll('.view-swap');
-  for (var i = 0; i < $viewSwapper.length; i++) {
-    if ($viewSwapper[i] === event.target) {
-      var dataView = $viewSwapper[i].getAttribute('data-view');
-      switchView(dataView);
-    }
+  if (!event.target.matches('.view-swap')) {
+    return;
+  }
+  if (event.target.matches('.view-swap')) {
```
Comment (codingcodymiller) [inline js/main.js]: You can remove this conditional and just call the function inside it directly. If you got past the previous conditional, that means the element does have the `view-swap` class, so you can just do the work. The magic of guards!!

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**